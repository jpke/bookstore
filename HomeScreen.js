import React from 'react';
import { StyleSheet, Text, TouchableHighlight, View } from 'react-native';

export default class HomeScreen extends React.Component {
  navigate(routeName) {
    console.log("routeName: ", routeName);
    this.props.navigator.push({
      name: routeName
    });
  }
  render() {
    return (
      <View style={styles.container}>
        <Text style={styles.title}>Welcome to the BookStore</Text>
        <View style={styles.bookList}>
          <Text style={styles.menuItem}>Your Books</Text>
          {this.props.books.map((book, index) => {
            return (
              <View key={index} style={styles.book}>
                <Text>{book.title}</Text>
                <Text>{book.author}</Text>
              </View>
            )
          })}
        </View>
        <View style={styles.menu}>
          <TouchableHighlight style={styles.button} onPress={() => this.navigate('OrderHistory')}>
            <Text style={styles.menuItem}>Orders</Text>
          </TouchableHighlight>
          <TouchableHighlight style={styles.button} onPress={() => this.navigate('Admin')}>
            <Text style={styles.menuItem}>Admin</Text>
          </TouchableHighlight>
        </View>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'space-around',
  },
  title: {
    fontSize: 25,
  },
  bookList: {
    height: 300,
    justifyContent: 'space-between',
  },
  book: {
    height: 20,
    alignItems: 'stretch',
    backgroundColor: '#ffcd67',
    padding: 5

  },
  menu: {
    height: 100,
    justifyContent: 'space-around',
    alignItems: 'center',
  },
  menuItem: {
    fontSize: 18,
    color: '#FFFFFF',
  },
  button: {
    backgroundColor: '#6799FF',
    width: 250,
    padding: 10,
    margin: 5,
    borderRadius: 10,
    alignItems: 'center',
  }
});
