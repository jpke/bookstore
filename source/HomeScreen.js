import React from 'react';
import { StyleSheet,
         Text,
         TouchableHighlight,
         View,
         ListView,
       } from 'react-native';

export default class HomeScreen extends React.Component {
  constructor(props) {
    super(props)
    const dataSource = new ListView.DataSource(
      {rowHasChanged: (r1, r2) => r1.title !== r2.title}
    )
    this.state = {
      dataSource: dataSource
    }
  }


  navigate(routeName) {
    this.props.navigator.push({
      name: routeName
    });
  }

  render() {
    return (
      <View style={styles.container}>
        <Text style={styles.title}>Pocket BookStore</Text>
        <Text style={styles.subTitle}>Select from Available Books</Text>
          <View style={styles.bookList}>

            <ListView
              dataSource={this.state.dataSource.cloneWithRows(this.props.books)}
              renderRow={(book) => {
                return (
                    <View>
                      <TouchableHighlight style={styles.rowContainer}
                          onPress={() => {
                            this.props.selectBook(book)
                            this.navigate('Order')
                          }}>
                        <View>
                          <Text style={styles.bookTitle}>{book.title}</Text>
                          <Text style={styles.bookAuthor}>{book.author}</Text>
                        </View>
                      </TouchableHighlight>
                      <View style={styles.separator}/>
                    </View>
                  )}}
            />

          </View>
        <View style={styles.menu}>
          <TouchableHighlight style={styles.button} onPress={() => this.navigate('OrderHistory')}>
            <Text style={styles.menuItem}>Orders</Text>
          </TouchableHighlight>
          <TouchableHighlight style={styles.button} onPress={() => this.navigate('Admin')}>
            <Text style={styles.menuItem}>Admin</Text>
          </TouchableHighlight>
        </View>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'space-around',
  },
  title: {
    marginTop: 45,
    fontSize: 35,
    flex: 1.5,
    color: '#ffcd67',
    textShadowColor: 'black',
    textShadowOffset: {width: 1, height: 1},
    textShadowRadius: 4
  },
  subTitle: {
    fontSize: 20,
    flex: 1,
    color: '#fff',
    textShadowColor: 'black',
    textShadowOffset: {width: 1, height: 1},
    textShadowRadius: 4
  },
  bookList: {
    flex: 6,
    width: 320,
  },
  rowContainer: {
    padding: 1,
    borderRadius: 5,
    marginTop: 5,
    marginBottom: 5,
  },
  bookTitle: {
    fontSize: 17,
    padding: 5,
    color: '#fff',
    textShadowColor: 'black',
    textShadowOffset: {width: 1, height: 1},
    textShadowRadius: 4
  },
  bookAuthor: {
    fontSize: 14,
    padding: 5,
    color: '#fff',
    textShadowColor: 'black',
    textShadowOffset: {width: 1, height: 1},
    textShadowRadius: 4
  },
  separator: {
    height: 1,
    backgroundColor: '#dddddd'
  },
  menu: {
    flex: 1.5,
    flexDirection: 'row',
    justifyContent: 'space-around',
    alignItems: 'center',
  },
  menuItem: {
    fontSize: 18,
    color: '#FFFFFF',
  },
  button: {
    backgroundColor: '#6799FF',
    width: 100,
    padding: 10,
    margin: 5,
    borderRadius: 10,
    alignItems: 'center',
  }
});
