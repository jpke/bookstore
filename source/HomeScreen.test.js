import React from 'react';
import ReactNative from 'react-native';
import HomeScreen from './HomeScreen';

import renderer from 'react-test-renderer';

function selectNewBook(newBook) {
  console.log("available book to order: ", newBook);
  this.setState({
    bookToOrder: newBook
  })
}

it('renders correctly', () => {
  const tree = renderer.create(
    <HomeScreen
      books={[
        {
          title: "The Evolution of Civilizations",
          author: "Carroll Quigley",
        },
        {
          title: "How the Mind Works",
          author: "Stephan Pinker",

        }
      ]}
      selectBook={selectNewBook}/>
  ).toJSON();
  expect(tree).toMatchSnapshot();
})
