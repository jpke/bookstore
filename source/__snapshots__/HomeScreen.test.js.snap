// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "space-around",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 2.2,
        "marginBottom": 10,
        "marginTop": 15,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "alignItems": "flex-end",
          "color": "#ffcd67",
          "flex": 1.5,
          "fontSize": 40,
          "textShadowColor": "black",
          "textShadowOffset": Object {
            "height": 1,
            "width": 1,
          },
          "textShadowRadius": 4,
        }
      }
    >
      Pocket BookStore
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#fff",
          "flex": 1,
          "fontSize": 30,
          "textShadowColor": "black",
          "textShadowOffset": Object {
            "height": 1,
            "width": 1,
          },
          "textShadowRadius": 4,
        }
      }
    >
      Select from Available Books
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 6,
        "width": 320,
      }
    }
  >
    <ScrollView
      style={
        Object {
          "flex": 6,
          "width": 320,
        }
      }
    >
      <View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              Object {
                "backgroundColor": "rgba(255,205,103,0.8)",
                "borderRadius": 5,
                "flex": 1,
                "marginBottom": 6,
                "marginTop": 6,
                "padding": 6,
              },
            ]
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#fff",
                  "fontSize": 22,
                  "padding": 5,
                  "textShadowColor": "black",
                  "textShadowOffset": Object {
                    "height": 1,
                    "width": 1,
                  },
                  "textShadowRadius": 4,
                }
              }
            >
              The Evolution of Civilizations
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 18,
                  "padding": 5,
                }
              }
            >
              Carroll Quigley
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "backgroundColor": "#dddddd",
              "height": 1,
            }
          }
        />
      </View>
      <View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              Object {
                "backgroundColor": "rgba(255,205,103,0.8)",
                "borderRadius": 5,
                "flex": 1,
                "marginBottom": 6,
                "marginTop": 6,
                "padding": 6,
              },
            ]
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#fff",
                  "fontSize": 22,
                  "padding": 5,
                  "textShadowColor": "black",
                  "textShadowOffset": Object {
                    "height": 1,
                    "width": 1,
                  },
                  "textShadowRadius": 4,
                }
              }
            >
              How the Mind Works
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 18,
                  "padding": 5,
                }
              }
            >
              Stephan Pinker
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "backgroundColor": "#dddddd",
              "height": 1,
            }
          }
        />
      </View>
    </ScrollView>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1.5,
        "flexDirection": "row",
        "justifyContent": "space-around",
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Object {
            "alignItems": "center",
            "backgroundColor": "#6799FF",
            "borderRadius": 10,
            "margin": 5,
            "padding": 10,
            "width": 100,
          },
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#FFFFFF",
            "fontSize": 18,
          }
        }
      >
        Orders
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Object {
            "alignItems": "center",
            "backgroundColor": "#6799FF",
            "borderRadius": 10,
            "margin": 5,
            "padding": 10,
            "width": 100,
          },
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#FFFFFF",
            "fontSize": 18,
          }
        }
      >
        Admin
      </Text>
    </View>
  </View>
</View>
`;
